compiler/testData/codegen/box/constants/constantsInWhen.kt

```diff
@@ -1,7 +1,6 @@
 // Even before any IR lowerings, the type of `when` is determined to be
 // Unit even though the outer `if` still returns `Int?`. This results
 // in a ClassCastException when that Unit is converted into a Number.
-// IGNORE_BACKEND: JVM_IR
```

Comment:
The comment above is probably also not needed after this change?

Reply:
Oops. Thanks for catching that!